testnet/stacks-node/src/neon_node.rs

```diff
@@ -1026,11 +1035,25 @@ impl InitializedNeonNode {
         ) {
             Some(vrfp) => vrfp,
             None => {
-                error!(
-                    "Failed to generate proof with {:?}",
-                    &registered_key.vrf_public_key
-                );
-                return None;
+                // Try to recover a key registered in a former session.
+                // registered_key.block_height gives us a pointer to the height of the block
+                // holding the key register op, but the VRF was derived using the height of one
+                // of the parents blocks.
+                let _ = keychain.rotate_vrf_keypair(registered_key.block_height - 1);
```

Comment:
I'm not sure what's going on here?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2073/commits/f0baf47900640401ab6f6dee385b6b7d7e25e89f